src/swipes.py

```diff
@@ -0,0 +1,304 @@
+from os.path import isfile
+from datetime import date, datetime, timedelta
+import json
+import numpy as np
+import pandas as pd
+
+from src.constants import (
+  BRB_ONLY,
+  DINING_HALL,
+  EATERY_DATA_PATH,
+  LOCATION_NAMES,
+  ISOLATE_COUNTER_SWIPES,
+  ISOLATE_DATE,
+  ISOLATE_SWIPES,
+  SCHOOL_BREAKS,
+  TRILLIUM,
+  WAIT_TIME_CONVERSION,
+  WEEKDAYS,
+)
+from src.eatery import string_to_date_range
+from src.types import SwipeDataType
+
+weekdays = {index: day for day, index in WEEKDAYS.items()}  # invert our weekday -> index dict
+breaks = {}
+for label, dates in SCHOOL_BREAKS.items():
+  breaks[label] = string_to_date_range(dates)
+
+def parse_to_csv(file_name='data.csv', limit=-1):
+  """
+  Takes in a data.log file of swipe logs and converts them to a tabular format.
+  Argument limit: used to limit number of logs to parse (for testing purposes).
+  Creates one csv file (updates file if already exists):
+  data.csv - stores all swipe events
+    date: date of swipe event, string (mm/dd/yyyy)
+    weekday: weekday of swipe event, string (ex: monday, tuesday, etc.)
+    in_session: whether or not school is on a break, boolean
+    location: eatery of swipe event, string (some locations have different names than dining.now's)
+    start_time: left edge of timeblock, string (hh:mm AM/PM)
+    end_time: right edge of timeblock, string (hh:mm AM/PM)
+    swipes: number of swipes, int
+    brb_only: indicates if this location is a brb only eatery (used for calcs), boolean
+    dining_hall: indicates if a location is a dining hall (used for calcs), boolean
+    trillium: indicates if a location is Trillium (used for calcs), boolean
+  """
+  global breaks
+  global weekdays
+  MAIN_CSV = '{}{}'.format(EATERY_DATA_PATH, file_name)
+  try:
+    with open('{}{}'.format(EATERY_DATA_PATH, 'data.log'), 'r') as swipe_data:
+      i = 0
+      line_counter = 0
+      data_list = []
+      update_info = {}
+
+      # check if data csv file already exists
+      if isfile(MAIN_CSV):
+        try:
+          df = pd.read_csv(MAIN_CSV)
+          marker_row = df.head(1)
+          print(marker_row.iloc[0]['date'])
+          df.drop(marker_row.index, inplace=True)  # remove previous time marker
+          update_info['end'] = marker_row
+          update_info['old_data'] = df
+          print('marker and old data found')
+        except Exception as e:
+          print('failed in marker row creation')
+
+      # read most recent data first
+      for line in list(reversed(swipe_data.readlines())):
```

Comment:
hm it's not good to call `readlines()`, since you don't want to overload memory, it's much better to iterate line by line!